homeassistant/components/hausbus/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Config flow for Haus-Bus integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from pyhausbus.BusDataMessage import BusDataMessage
+from pyhausbus.de.hausbus.homeassistant.proxy.controller.data.ModuleId import ModuleId
+from pyhausbus.HomeServer import HomeServer
+from pyhausbus.IBusDataListener import IBusDataListener
+from pyhausbus.ObjectId import ObjectId
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_SCHEMA = vol.Schema({})
+
+
+class ConfigFlow(IBusDataListener, config_entries.ConfigFlow, domain=DOMAIN):  # type: ignore[misc]
+    """Handle a config flow for hausbus."""
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._found_device = False
+        self._search_task: asyncio.Task | None = None
+        self.home_server = HomeServer()
+        self.home_server.addBusEventListener(self)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            # start searching for devices
+            return await self.async_step_wait_for_device()
+
+        return self.async_show_form(
+            step_id="user", data_schema=STEP_USER_SCHEMA, errors=errors
+        )
+
+    async def async_step_wait_for_device(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Wait for a hausbus device to be found."""
+        if not self._search_task:
+            self._search_task = self.hass.async_create_task(
+                self._async_wait_for_device()
+            )
+
+        if not self._search_task.done():
+            return self.async_show_progress(
+                step_id="wait_for_device",
+                progress_action="wait_for_device",
```

Comment:
There are no translations for the step_id + action, please add that, otherwise I think the user only sees a spinner with no explanation while searching.

Reply:
Are there any resources regarding how to correctly use translations? I found it hard to find the correct keys in the strings.json.

Reply:
There's some explanations here: https://developers.home-assistant.io/docs/data_entry_flow_index#labels--descriptions

When you test locally, you need to make sure `python3 -m script.translations develop --all` is executed before `hass`.
Also, you may need to force-refresh your browser for it to re-fetch translations.